mesonbuild/dependencies/python.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2022 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+from pathlib import Path
+import sysconfig
+import typing as T
+
+from .. import mlog
+from .base import DependencyMethods, SystemDependency
+from .factory import DependencyFactory
+from ..environment import detect_cpu_family
+
+if T.TYPE_CHECKING:
+    from ..environment import Environment
+
+
+class Python3DependencySystem(SystemDependency):
+    def __init__(self, name: str, environment: 'Environment', kwargs: T.Dict[str, T.Any]) -> None:
```

Comment:
I see that you are just moving code around and I don't know what the the Meson coding style says about this, but if you have `from __futute__ import annotations` using string for the annotations is redundant as they are not resolved when not type checking anyway. Namely, this could be `environment: Environment`.